R/regressionlinear.R

```diff
@@ -1419,115 +1485,145 @@ RegressionLinear <- function(jaspResults, dataset = NULL, options) {
-    
-    sePred <- sqrt(sum(res^2) / dfRes) * 
-      sqrt(1 + 1 / length(res) + (xVar - mean(xVar))^2 / sum((xVar - mean(xVar))^2))
-    
-    ciPred <- 1 - (1 - predictionIntervalsInterval)/2
-    
-    upperPredInt <- (intercept + slope * xVar) + qt(ciPred, dfRes) *  sePred
-    lowerPredInt <- (intercept + slope * xVar) - qt(ciPred, dfRes) *  sePred
-    
-    p <- p + ggplot2::geom_line(data = data.frame(x = xVar,
-                                                  y = upperPredInt),
-                                mapping = ggplot2::aes(x = x, y = y), 
-                                col = "darkgreen", linetype = "longdash", size = 1) + 
-      ggplot2::geom_line(data = data.frame(x = xVar,
-                                           y = lowerPredInt),
-                                mapping = ggplot2::aes(x = x, y = y), 
-                                col = "darkgreen", linetype = "longdash", size = 1)
-    
-  }
-  
-  p <- jaspGraphs::drawPoints(p, dat = data.frame(x = xVar, y = res), size = 3)
-
-  # JASP theme
-  p <- jaspGraphs::themeJasp(p)
+    if (confidenceIntervals) {
+
+      seConf <- sqrt(sum(res^2) / dfRes) *
+        sqrt(1 / length(res) + (xVar - mean(xVar))^2 / sum((xVar - mean(xVar))^2))
+
+      ciConf <- 1 - (1 - confidenceIntervalsInterval)/2
+
+      upperConfInt <- (intercept + slope * xVar) + qt(ciConf, dfRes) * seConf
+      lowerConfInt <- (intercept + slope * xVar) - qt(ciConf, dfRes) * seConf
+
+      # ggplot2::geom_errorbar()
+
+      confidenceIntervalLines <- if (is.factor(xVar)) {
+        ggplot2::geom_errorbar(
+          data = data.frame(x = levels(xVar), ymax = upperConfInt, ymin = lowerConfInt),
+          mapping = ggplot2::aes(x = x, ymax = ymax, ymin = ymin),
+          colour = "darkblue", linetype = "dashed"
+        )
+      } else {
+        ggplot2::geom_line(
+          data = data.frame(x = xVar, y = c(upperConfInt, lowerConfInt), g = rep(1:2, c(length(upperConfInt), length(lowerConfInt)))),
+          mapping = ggplot2::aes(x = x, y = y, group = g),
+          colour = "darkblue", linetype = "dashed"
+        )
+      }
+
+    }
+
+    if (predictionIntervals) {
+
+      sePred <- sqrt(sum(res^2) / dfRes) *
+        sqrt(1 + 1 / length(res) + (xVar - mean(xVar))^2 / sum((xVar - mean(xVar))^2))
+
+      ciPred <- 1 - (1 - predictionIntervalsInterval)/2
+
+      upperPredInt <- (intercept + slope * xVar) + qt(ciPred, dfRes) * sePred
+      lowerPredInt <- (intercept + slope * xVar) - qt(ciPred, dfRes) * sePred
+
+      predictionIntervalLines <- if (is.factor(xVar)) {
+        ggplot2::geom_errorbar(
+          data = data.frame(x = levels(xVar), ymax = upperPredInt, ymin = lowerPredInt),
+          mapping = ggplot2::aes(x = x, ymax = ymax, ymin = ymin),
+          colour = "darkgreen", linetype = "longdash"
+        )
+      } else {
+        ggplot2::geom_line(data = data.frame(x = xVar, y = c(upperConfInt, lowerConfInt), g = rep(1:2, c(length(upperConfInt), length(lowerConfInt)))),
```

Comment:
```suggestion
        ggplot2::geom_line(data = data.frame(x = xVar, y = c(upperPredInt, lowerPredInt), g = rep(1:2, c(length(upperPredInt), length(lowerPredInt)))),
```

Reply:
Forgot to change confidence -> prediction intervals, so that now you draw the same interval twice.

Reply:
I wonder why this was not caught by the unit test though, I though @maltelueken added new tests for prediction and confidence intervals in this PR: https://github.com/jasp-stats/jasp-desktop/pull/4220...

Ah I see, the change to the unit test was not copied to this module though: https://github.com/jasp-stats/jaspRegression/commit/9ec4037fdffa42d26d4bf46b38239c831e52af29

Would be good to add it back I think. @vandenman do you wanna do it here or should we just tackle this later?

Reply:
Good catch! I was already wondering why the two lines were the same.. I'll add the unit test back in this PR.

Reply:
actually, it looks like there are some tests for these plots already:

https://github.com/jasp-stats/jasp-desktop/blob/22c87484bb28c7eb3b5ace15ae269942a356d7d5/JASP-Tests/R/tests/testthat/test-regressionlinear.R#L444-L456

and also here:

https://github.com/jasp-stats/jaspRegression/blob/f85592d261a095cbc0c2279611c49288965779b8/tests/testthat/test-regressionlinear.R#L444-L456

in both stable and this module, the unit tests files are identical for linear regression... In my next commit I'll re-enabled some of the partial plots, but these are already commented out in stable.